controllers/spacerequest/spacerequest_controller.go

```diff
@@ -399,6 +412,125 @@ func (r *Reconciler) deleteExistingSubSpace(logger logr.Logger, subSpace *toolch
 	return true, nil
 }
 
+func (r *Reconciler) ensureSecretForProvisionedNamespaces(logger logr.Logger, memberClusterWithSpaceRequest cluster.Cluster, spaceRequest *toolchainv1alpha1.SpaceRequest, subSpace *toolchainv1alpha1.Space) error {
+	if len(subSpace.Status.ProvisionedNamespaces) == 0 {
+		logger.Info("provisioned namespaces not available yet...")
+		return nil
+	}
+	logger.Info("ensure secret for provisioned namespaces")
+
+	var namespaceAccess []toolchainv1alpha1.NamespaceAccess
+	for _, namespace := range subSpace.Status.ProvisionedNamespaces {
+		// check if kubeconfig secret exists,
+		// if it doesn't exist it will be created
+		secretList := &v1.SecretList{}
+		secretLabels := client.MatchingLabels{
+			toolchainv1alpha1.SpaceRequestLabelKey:                     spaceRequest.GetName(),
+			toolchainv1alpha1.SpaceRequestNamespaceLabelKey:            spaceRequest.GetNamespace(),
+			toolchainv1alpha1.SpaceRequestProvisionedNamespaceLabelKey: namespace.Name,
+		}
+		if err := memberClusterWithSpaceRequest.Client.List(context.TODO(), secretList, secretLabels, client.InNamespace(spaceRequest.GetNamespace())); err != nil {
+			return errs.Wrap(err, fmt.Sprintf(`attempt to list Secrets associated with spaceRequest %s in namespace %s failed`, spaceRequest.GetName(), spaceRequest.GetNamespace()))
+		}
+
+		kubeConfigSecret := &v1.Secret{}
+		switch {
+		case len(secretList.Items) == 0:
+			// create the secret for this namespace
+			clientConfig, err := r.generateKubeConfig(memberClusterWithSpaceRequest, namespace.Name)
+			if err != nil {
+				return err
+			}
+			clientConfigFormatted, err := clientcmd.Write(*clientConfig)
+			if err != nil {
+				return err
+			}
+			kubeConfigSecret = &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: spaceRequest.Name + "-",
+					Namespace:    spaceRequest.Namespace,
+					Labels:       secretLabels,
+				},
+				Type: v1.SecretTypeOpaque,
+				StringData: map[string]string{
+					"kubeconfig": string(clientConfigFormatted),
+				},
+			}
+			if err := controllerutil.SetControllerReference(spaceRequest, kubeConfigSecret, r.Scheme); err != nil {
+				return errs.Wrap(err, "error setting controller reference for secret "+kubeConfigSecret.Name)
+			}
+			if err := memberClusterWithSpaceRequest.Client.Create(context.TODO(), kubeConfigSecret); err != nil {
+				return err
+			}
+			logger.Info("Created Secret", "Name", kubeConfigSecret.Name)
+
+			// add provisioned namespace and name of the secret that provides access to the namespace
+			namespaceAccess = append(namespaceAccess, toolchainv1alpha1.NamespaceAccess{
+				Name:      namespace.Name,
+				SecretRef: kubeConfigSecret.Name,
+			})
+
+		case len(secretList.Items) == 1:
+			// a secret is already present for this namespace
+			namespaceAccess = append(namespaceAccess, toolchainv1alpha1.NamespaceAccess{
+				Name:      namespace.Name,
+				SecretRef: secretList.Items[0].Name,
+			})
```

Comment:
I don't see any test for this use-case

Reply:
Good catch ! It should be now covered in https://github.com/codeready-toolchain/host-operator/commit/03a04d98122bf138479a7714e44f3ea2db56f826